src/pages/index.js

```diff
@@ -5,52 +5,293 @@ import PopupWithForm from "../components/PopupWithForm.js";
+      card.handleDeleteClick(); // Assuming this method exists to remove the card from the DOM
+    })
+    .catch((error) => {
+      console.error("Error deleting card:", error);
+    })
+    .finally(() => {
+      deleteSubmitButton.textContent = "Yes"; // Reset button text after completion
+    });
+}
+
+// ----------------------------
+
+// like button
+// Function to handle like click
+
+function findCardById(cardId) {
+  return initialCards.find((card) => card._id === cardId || card.id === cardId);
+}
+function handleLikeClick(cardData) {
+  if (cardData._isLiked) {
+    api
+      .removeLike(cardData._id)
+      .then((updatedCard) => {
+        cardData.setLike(false); // Update local like status
+      })
+      .catch((err) => console.log(err));
+  } else {
+    api
+      .addLike(cardData._id)
+      .then((updatedCard) => {
+        cardData.setLike(true); // Update local like status
+      })
+      .catch((err) => console.log(err));
+  }
+  // cardData.renderLikes();
+}
+
+// Function to update card likes in the UI
+function updateCardLikes(cardId, likes) {
+  const cardElement = document.querySelector(`.card[data-id="${cardId}"]`); // Make sure card elements have data-id attribute
+  console.log("cardElement:", cardElement); // Log cardElement
+
+  // Check if cardElement is found
+  if (cardElement === null) {
+    console.error(`Card element not found for id: ${cardId}`);
+    return;
+  }
+  const likeCountElement = cardElement.querySelector(".card__like-count");
+
+  likeCountElement.textContent = likes.length;
+
+  const likeButton = cardElement.querySelector(".card__button-like");
+  if (likes.some((user) => user._id === currentUser._id)) {
+    likeButton.classList.add("card__button-like_active");
+  } else {
+    likeButton.classList.remove("card__button-like_active");
+  }
+}
+
+// ----------------------------
+
+const profileSubmitButton = document.querySelector("#profile-submit-button");
+
+function handleProfileFormSubmit(data) {
+  // event.preventDefault();
+  profileSubmitButton.textContent = "Saving...";
+
+  const formData = new FormData(profileForm);
+
+  const formValues = {
+    name: formData.get("title"),
+    about: formData.get("description"),
+    avatar: formData.get("avatar"),
+  };
+
+  api
+    .setUserInfo(formValues)
+    .then((data) => {
+      userInfo.setUserInfo({
+        name: data.name,
+        job: data.about,
+        avatar: data.avatar,
+      });
+      editProfilePopup.close();
+      // this._popupForm.reset();
+    })
+    .catch((error) => console.error("Error updating profile:", error))
+    .finally(() => {
+      profileSubmitButton.textContent = "Save";
+    });
+}
+function updateProfileDescription(data) {
+  document.querySelector(".name").textContent = data.name;
+  document.querySelector(".about").textContent = data.about;
+  document.querySelector(".avatar").src = data.avatar;
+}
+
+// Fetch user info and update the DOM
+api.getUserInfo().then((data) => {
```

Comment:
Please, delete `api.getUserInfo()` . You have everything in line 59 with `userInfo.setUserInfo(userData);`